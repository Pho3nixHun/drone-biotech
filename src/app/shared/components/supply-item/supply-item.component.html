<div class="flex flex-col">
    <ng-content select="[h3, p]" />
</div>
