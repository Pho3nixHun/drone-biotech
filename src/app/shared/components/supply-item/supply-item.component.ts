import { Component } from '@angular/core';

/**
 * SupplyItemComponent
 *
 * Type: Presentational
 *
 * Scope:
 * - Responsible for rendering an individual Supply Item.
 * - Manages the styling and layout necessary for the item's presentation.
 *
 * Out-of-Scope:
 * - Should not handle data fetching.
 * - Contains no business logic related to the item.
 * - Does not manage or transform the item's data.
 *
 * Purpose (optional):
 * To visually present an individual supply item.
 */
@Component({
    selector: 'app-supply-item',
    imports: [],
    templateUrl: './supply-item.component.html',
})
export class SupplyItemComponent {}
