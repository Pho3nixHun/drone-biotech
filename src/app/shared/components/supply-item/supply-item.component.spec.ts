import { ComponentFixture, TestBed } from '@angular/core/testing';
import { SupplyItemComponent } from './supply-item.component';
import { Component } from '@angular/core';

@Component({
    imports: [SupplyItemComponent],
    template: `
        <app-supply-item>
            <h3>Should be projected</h3>
            <p>Should be projected</p>
            <div>Should not be projected</div>
        </app-supply-item>
    `,
})
class TestHostComponent {}

describe('SupplyItemComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should project the content and render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
