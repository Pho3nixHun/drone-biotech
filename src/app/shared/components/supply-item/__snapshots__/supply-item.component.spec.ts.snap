// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplyItemComponent should project the content and render the template correctly 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <app-supply-item>
    <div
      class="flex flex-col"
    >
      <h3>
        Should be projected
      </h3>
      <p>
        Should be projected
      </p>
    </div>
  </app-supply-item>
</div>
`;
